crates/solver/src/liquidity.rs

```diff
@@ -90,40 +90,50 @@ pub enum Exchange {
 }
 
 /// Used to differentiate between different types of orders that can be sent to solvers.
-/// User orders containing OrderUid are the orders from the orderbook.
+/// User orders (market + limit) containing OrderUid are the orders from the orderbook.
 #[derive(Debug, Clone)]
 #[cfg_attr(test, derive(Derivative))]
 #[cfg_attr(test, derivative(PartialEq))]
 pub enum LimitOrderUid {
-    OrderUid(OrderUid),
+    Market(OrderUid),
+    Limit(OrderUid),
+    Liquidity(LiquidityOrderUid),
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Derivative))]
+#[cfg_attr(test, derivative(PartialEq))]
+pub enum LiquidityOrderUid {
+    /// Originates from the auction model of solvable orders
+    User(OrderUid),
```

Comment:
Maybe some other name? `WhitelistedUser` maybe?

Reply:
I think `CowProtocol` or just `Protocol` to indicate it is a protocol limit order would be good.

I also think `User` is a bit general.

Reply:
I've changed it to ProtocolOrForeign, added comment why